ivadomed/testing.py

```diff
@@ -254,6 +254,13 @@ def run_inference(test_loader, model, model_params, testing_params, ofolder, cud
                         #                              fname_pred.split(".nii.gz")[0] + '_color.nii.gz',
                         #                              imed_utils.AXIS_DCT[testing_params['slice_axis']])
 
+                    # For Microscopy PNG/TIF files (TODO: implement OMETIFF behavior)
+                    extension = imed_loader_utils.get_file_extension(fname_ref)
+                    if "nii" not in extension and fname_pred:
+                        output_list = imed_inference.split_classes(output_nii)
+                        imed_inference.pred_to_png(output_list,
+                                                   testing_params['target_suffix'],
+                                                   fname_pred.split("_pred.nii.gz")[0])
```

Comment:
One curious thing I noticed in the `test` output was this, though:

```
2021-07-06 13:54:16.147 | INFO     | ivadomed.testing:test:51 - Loading model: new_loader/best_model.pt
Inference - Iteration 0:   0%|          | 0/1 [00:00<?, ?it/s]Lossy conversion from float64 to uint8. Range [0, 1]. Convert image to uint8 prior to saving to suppress this warning.
2021-07-06 13:54:23.388 | WARNING  | ivadomed.testing:run_inference:248 - No color labels saved due to a temporary issue. For more details see:https://github.com/ivadomed/ivadomed/issues/720
Lossy conversion from float64 to uint8. Range [0, 1]. Convert image to uint8 prior to saving to suppress this warning.
2021-07-06 13:54:23.646 | WARNING  | ivadomed.testing:run_inference:248 - No color labels saved due to a temporary issue. For more details see:https://github.com/ivadomed/ivadomed/issues/720
```

(The error message line of code is just a few lines above this, but GH won't let me highlight.)

It's out of scope for this PR, but I realize the linked issue https://github.com/ivadomed/ivadomed/issues/720 was apparently closed a few days ago -- can we update this code?

Reply:
Good catch! I had not realized that #720 had been addressed. It is indeed out of scope but I realized that the function was using the variable `fname_tmp` which no longer exists and was replaced by `fname_ref`. I made the change in d861fe9.

However, I read a little bit the PR #832 and I'm not sure if the functionality is fixed based on this [comment](https://github.com/ivadomed/ivadomed/pull/832#pullrequestreview-695691581), I tested it by uncommenting `imed_visualize.save_color_labels` and got the error blow so I think it is not fixed (tagging @dyt811 so he can follow-up).
Thanks again!

<details>
<summary> Error  </summary>

```
Traceback (most recent call last):
  File "/home/mhbourget/venv-ivadomed-38c/bin/ivadomed", line 11, in <module>
    load_entry_point('ivadomed', 'console_scripts', 'ivadomed')()
  File "/home/mhbourget/code/ivadomed/ivadomed/main.py", line 566, in run_main
    run_command(context=context,
  File "/home/mhbourget/code/ivadomed/ivadomed/main.py", line 470, in run_command
    pred_metrics = imed_testing.test(model_params=model_params,
  File "/home/mhbourget/code/ivadomed/ivadomed/testing.py", line 76, in test
    preds_npy, gt_npy = run_inference(test_loader, model, model_params, testing_params, path_3Dpred,
  File "/home/mhbourget/code/ivadomed/ivadomed/testing.py", line 251, in run_inference
    imed_visualize.save_color_labels(np.stack(pred_tmp_lst, -1),
  File "/home/mhbourget/code/ivadomed/ivadomed/visualize.py", line 101, in save_color_labels
    imed_inference.pred_to_nib([multi_labeled_pred], [], gt_filename,
  File "/home/mhbourget/code/ivadomed/ivadomed/inference.py", line 205, in pred_to_nib
    nib.save(nib_pred, fname_out)
  File "/home/mhbourget/venv-ivadomed-38c/lib/python3.8/site-packages/nibabel/loadsave.py", line 99, in save
    img.to_filename(filename)
  File "/home/mhbourget/venv-ivadomed-38c/lib/python3.8/site-packages/nibabel/filebasedimages.py", line 333, in to_filename
    self.to_file_map()
  File "/home/mhbourget/venv-ivadomed-38c/lib/python3.8/site-packages/nibabel/analyze.py", line 1019, in to_file_map
    arr_writer = make_array_writer(data,
  File "/home/mhbourget/venv-ivadomed-38c/lib/python3.8/site-packages/nibabel/arraywriters.py", line 790, in make_array_writer
    return SlopeInterArrayWriter(data, out_type, **kwargs)
  File "/home/mhbourget/venv-ivadomed-38c/lib/python3.8/site-packages/nibabel/arraywriters.py", line 521, in __init__
    super(SlopeInterArrayWriter, self).__init__(array,
  File "/home/mhbourget/venv-ivadomed-38c/lib/python3.8/site-packages/nibabel/arraywriters.py", line 303, in __init__
    self.calc_scale()
  File "/home/mhbourget/venv-ivadomed-38c/lib/python3.8/site-packages/nibabel/arraywriters.py", line 351, in calc_scale
    if not self.scaling_needed():
  File "/home/mhbourget/venv-ivadomed-38c/lib/python3.8/site-packages/nibabel/arraywriters.py", line 325, in scaling_needed
    if not super(SlopeArrayWriter, self).scaling_needed():
  File "/home/mhbourget/venv-ivadomed-38c/lib/python3.8/site-packages/nibabel/arraywriters.py", line 125, in scaling_needed
    raise WriterError('Cannot cast to or from non-numeric types')
nibabel.arraywriters.WriterError: Cannot cast to or from non-numeric types
```
</details>

Reply:
> Good catch! I had not realized that #720 had been addressed. It is indeed out of scope but I realized that the function was using the variable `fname_tmp` which no longer exists and was replaced by `fname_ref`. I made the change in [d861fe9](https://github.com/ivadomed/ivadomed/commit/d861fe9c25fe6ef2b35f45822ee71e904bbd810a).
> 
> However, I read a little bit the PR #832 and I'm not sure if the functionality is fixed based on this [comment](https://github.com/ivadomed/ivadomed/pull/832#pullrequestreview-695691581), I tested it by uncommenting `imed_visualize.save_color_labels` and got the error below so I think it is not fixed (tagging @dyt811 so he can follow-up).
> Thanks again!
> 
> Error

how did you test save_color_labels? you have to provide a gt_file with the right header (ie rgb data type and etc)

Reply:
I was not able to test the function because the gt_file doesn't have the right header, but I changed the variable name from `fname_tmp` to `fname_ref` anyway because this is part of a refactoring I did in #835 (but I had forgotten to change it in the commented code).
So the code for `imed_visualize.save_color_labels` is still commented, the function would need to be adjusted to match the header of our gt_file.